src/diff/props.js

```diff
@@ -117,6 +117,7 @@ export function setProperty(dom, name, value, oldValue, namespace) {
 			name != 'colSpan' &&
 			name != 'role' &&
 			name != 'popover' &&
+			!dom.constructor.observedAttributes?.includes(name) &&
```

Comment:
I'd be surprised if this wasn't breaking and I don't think it's inline with our existing flow here.

Preact doesn't make any attempt to guess what the user wants; we set props if the name is recognized (with a few exceptions), else, it's an attribute. This breaks that by attempting to guess the user wants an attribute instead.

#4419 is probably a better option, takes the guesswork out for Preact and gives the user more power.